sdks/python/apache_beam/runners/interactive/sql/beam_sql_magics.py

```diff
@@ -194,45 +220,72 @@ def beam_sql(self, line: str, cell: Optional[str] = None) -> Optional[PValue]:
             pcoll.element_type)
         return
       register_coder_for_schema(pcoll.element_type, verbose=verbose)
-
-    output_name, output = apply_sql(query, output_name, found)
+      # Only care about schemas defined by the user in the main module.
+      if hasattr(main_session, pcoll.element_type.__name__):
+        schemas.add(pcoll.element_type)
+
+    if runner and runner != 'DirectRunner':
```

Comment:
Why do we condition based on if the runner is the DirectRunner?

Reply:
DirectRunner is equivalent to not setting the argument at all. It means the SQL is executed locally.

Reply:
Let me switch the order of conditions to make it cleaner.